python/src/equistore/operations/unique_metadata.py

```diff
@@ -0,0 +1,245 @@
+"""
+Module for finding unique metadata for TensorMaps and TensorBlocks
+"""
+from typing import List, Optional, Tuple, Union
+
+import numpy as np
+
+from ..block import TensorBlock
+from ..labels import Labels
+from ..tensor import TensorMap
+
+
+def unique_metadata(
+    tensor: TensorMap,
+    axis: str,
+    names: Union[List[str], Tuple[str], str],
+    gradient_param: Optional[str] = None,
+) -> Labels:
+    """
+    Returns a :py:class:`Labels` object containing the unique metadata across
+    all blocks of the input :py:class:`TensorMap`  ``tensor``, for the specified
+    ``axis`` (either "samples" or "properties") and metatdata ``names``.
```

Comment:
Maybe a bit shorter sentence

```suggestion
    Returns a :py:class:`Labels` object containing the unique metadata across
    all blocks of the input :py:class:`TensorMap`  ``tensor``. Unique Labels are
    returned for the specified ``axis`` (either "samples" or "properties") and
    metatdata ``names``.
```